src/tls_stream.rs

```diff
@@ -620,10 +620,18 @@ impl<S> TlsStream<S>
         let mut out = 0;
         while self.out_buf.position() as usize != self.out_buf.get_ref().len() {
             let position = self.out_buf.position() as usize;
-            let nwritten = try!(self.stream.write(&self.out_buf.get_ref()[position..]));
+            let nwritten = self.stream.write(&self.out_buf.get_ref()[position..])?;
             out += nwritten;
             self.out_buf.set_position((position + nwritten) as u64);
         }
+        // TODO: Should unused memory in out_buf be freed up here?
```

Comment:
This will not cause any memory to be freed. We don't want to try to shrink out_buf anyway - it'll cap out at the maximum size of a TLS record. Reallocating on every write is pretty terrible perf-wise.